freqtrade/persistence/models.py

```diff
@@ -278,6 +297,75 @@ def open_date_utc(self):
     def close_date_utc(self):
         return self.close_date.replace(tzinfo=timezone.utc)
 
+    @property
+    def enter_side(self) -> str:
+        if self.is_short:
+            return "sell"
+        else:
+            return "buy"
+
+    @property
+    def exit_side(self) -> str:
+        if self.is_short:
+            return "buy"
+        else:
+            return "sell"
+
+    def __init__(self, **kwargs):
+        for key in kwargs:
+            setattr(self, key, kwargs[key])
+        if self.isolated_liq:
+            self.set_isolated_liq(self.isolated_liq)
+        self.recalc_open_trade_value()
+
+    def set_stop_loss(self, stop_loss: float):
+        """
+            Method you should use to set self.stop_loss.
+            Assures stop_loss is not passed the liquidation price
+        """
+        if self.isolated_liq is not None:
+            if self.is_short:
+                sl = min(stop_loss, self.isolated_liq)
+            else:
+                sl = max(stop_loss, self.isolated_liq)
+        else:
+            sl = stop_loss
+
+        if not self.stop_loss:
+            self.initial_stop_loss = sl
+        self.stop_loss = sl
+
+    def set_isolated_liq(self, isolated_liq: float):
+        """
+            Method you should use to set self.liquidation price.
+            Assures stop_loss is not passed the liquidation price
+        """
+        if self.stop_loss is not None:
+            if self.is_short:
+                self.stop_loss = min(self.stop_loss, isolated_liq)
+            else:
+                self.stop_loss = max(self.stop_loss, isolated_liq)
+        else:
+            self.initial_stop_loss = isolated_liq
+            self.stop_loss = isolated_liq
+
+        self.isolated_liq = isolated_liq
+
+    def set_is_short(self, is_short: bool):
+        self.is_short = is_short
+        self.recalc_open_trade_value()
```

Comment:
What's the purpose of this `set_is_short()` method?

I'd assume that the side (short / long) is determined at trade open - so i don't see this as something that can magically "change" later (you either bought - or you sold). You should (must!) know the side of the trade before you create a trade object.

Reply:
I guess it was mostly for testing, just to make sure that `self.recalc_open_trade_value()` was always called after